examples/sfml/tilesheet.rs

```diff
@@ -0,0 +1,73 @@
+use std::path::Path;
+
+use sfml::{
+    graphics::{FloatRect, IntRect, Texture},
+    SfBox,
+};
+use tiled::tileset::Tileset;
+
+/// A container for a tileset and the texture it references.
+pub struct Tilesheet {
+    texture: SfBox<Texture>,
+    tileset: Tileset,
+}
+
+impl Tilesheet {
+    /// Create a tilesheet from a Tiled tileset, loading its texture along the way.
+    pub fn from_tileset<'p>(tileset: Tileset) -> Self {
+        let tileset_image = tileset.image.as_ref().unwrap();
+
+        let texture = {
+            let texture_path = Path::new(&tileset_image.source);
+
+            Texture::from_file(texture_path.to_str().expect("obtaining valid UTF-8 path")).unwrap()
+        };
+
+        Tilesheet { texture, tileset }
+    }
+
+    pub fn texture(&self) -> &Texture {
+        &self.texture
+    }
+
+    pub fn tile_rect(&self, gid: u32) -> Option<IntRect> {
+        if gid == 0 {
+            return None;
+        }
+        let id = gid - self.tileset.first_gid;
```

Comment:
Since this is the `Tilesheet` class, I think it would be better if a local tile index was passed in. We should avoid proliferating the `gid` to this level. Then this function shouldn't need to return `Option` either.

Reply:
In my changes, I've removed the 0 check, but kept the gid for now. We can revisit that when we adjust the example to support multiple tilesets, because in that case we need to find a nice way to call `tile_rect` on the right `Tilesheet` instance, and this also involves using the `first_gid`. And... we might resolve that during loading anyway.